packages/typescript/internal/ts_project.bzl

```diff
@@ -668,13 +696,13 @@ def ts_project_macro(
     typing_maps_outs = []
 
     if not emit_declaration_only:
-        js_outs.extend(_out_paths(srcs, out_dir, root_dir, False, ".js"))
+        js_outs.extend(_out_paths(srcs, out_dir, root_dir, False, resolve_json_module, ".js"))
     if source_map and not emit_declaration_only:
-        map_outs.extend(_out_paths(srcs, out_dir, root_dir, False, ".js.map"))
+        map_outs.extend(_out_paths(srcs, out_dir, root_dir, False, False, ".js.map"))
     if declaration or composite:
-        typings_outs.extend(_out_paths(srcs, typings_out_dir, root_dir, allow_js, ".d.ts"))
+        typings_outs.extend(_out_paths(srcs, typings_out_dir, root_dir, allow_js, False, ".d.ts"))
```

Comment:
I think this is maybe not quite right. We need the json files even if emit_declaration_only is true. I think it might make more sense to put these in typings_outs?